### ISSUE TITLE
//// Comments

### ISSUE THREAD

**Issue by Foxtrek64** [DotNetAnalyzers/StyleCopAnalyzers#3924]
Hi all,

There seems to be a behavior within StyleCop.Analyzers that introduces a new comment type, `////`. This has the affect of performing inline suppressions of SA1005, SA1515 and potentially others. I have a very strong suspicion that this behavior is the result of a bug, however it does introduce an interesting feature. For example:

```cs
A();
// B(); // SA1515: Single-line comment should be preceeded by blank line.
C();

A();
//// B(); // SA1515 is not raised.
C();

A();
//B(); // SA1005: Single line comment should begin with a space.
C();

A();
////B(); // SA1005 is not raised.
C();
```

Or as a more practical scenario, as suggested in #3923, you can use it to suppress SA1515 in this scenario:
```cs
public static Person FromArray(string[] values)
{
    return new Person
    (
        FirstName: values[0],
        LastName: values[1],
        //// SSN: values[2],
        Age: values[4]
    );
}
```

I am not sure what analyzers it suppresses, but SA1005 and SA1515 are among them. I have not conducted further testing.

How should we handle this? Should this be a documented and supported solution moving forward or do we want to disallow this and require people to use `#pragma warning disable/restore` or to disable the analyzer globally?

**Comment by Foxtrek64** [comment]
Found this is actually documented in SA1515's info. Not sure how I missed that.
